app/modules/ph/endpoints_ph.py

```diff
@@ -0,0 +1,263 @@
+    save_file_as_data,
+    save_pdf_first_page_as_image,
+)
+
+module = Module(
+    root="ph",
+    tag="ph",
+    default_allowed_groups_ids=[GroupType.student],
+)
+
+
+@module.router.get(
+    "/ph/{paper_id}/pdf",
+    response_class=FileResponse,
+    status_code=200,
+)
+async def get_paper_pdf(
+    paper_id: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    paper = await cruds_ph.get_paper_by_id(db=db, paper_id=paper_id)
+    if paper is None:
+        raise HTTPException(
+            status_code=404,
+            detail="The paper does not exist.",
+        )
+
+    return get_file_from_data(
+        default_asset="assets/pdf/default_ph.pdf",
+        directory="ph/pdf",
+        filename=str(paper_id),
+    )
+
+
+@module.router.get(
+    "/ph/",
+    response_model=list[schemas_ph.PaperComplete],
+    status_code=200,
+)
+async def get_papers(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Return all editions until now, sorted from the latest to the oldest
+    """
+    result = await cruds_ph.get_papers(
+        db=db,
+        end_date=datetime.now(tz=UTC).date(),
+    )  # Return papers from the latest to the oldest until now
+    return result
+
+
+@module.router.get(
+    "/ph/admin",
+    response_model=list[schemas_ph.PaperComplete],
+    status_code=200,
+)
+async def get_papers_admin(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+):
+    """
+    Return all editions, sorted from the latest to the oldest
+    """
+    result = await cruds_ph.get_papers(
+        db=db,
+    )  # Return all papers from the latest to the oldest
+    return result
+
+
+@module.router.post(
+    "/ph/",
+    response_model=schemas_ph.PaperComplete,
+    status_code=201,
+)
+async def create_paper(
+    paper: schemas_ph.PaperBase,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    notification_tool: NotificationTool = Depends(get_notification_tool),
+):
+    """Create a new paper."""
+
+    paper_complete = schemas_ph.PaperComplete(
+        id=str(uuid.uuid4()),
+        **paper.model_dump(),
+    )
+    try:
+        paper_db = models_ph.Paper(
+            id=paper_complete.id,
+            name=paper_complete.name,
+            release_date=paper_complete.release_date,
+        )
+
+        now = datetime.now(UTC)
+
+        if paper_db.release_date >= now.date() - timedelta(days=30):
+            message = Message(
+                context=f"ph-{paper_db.id}",
+                is_visible=True,
+                title=f"📗 PH - {paper_db.name}",
+                content="A new paper has been released! 🎉",
+                delivery_datetime=datetime.fromisoformat(
+                    paper_db.release_date.isoformat(),
+                ),
```

Comment:
Maybe we could send the notification during the day, instead of midnight
```suggestion
                delivery_datetime=datetime.combine(paper_db.release_date, time(hour=8, tzinfo=UTC)),
```